test/e2e/ctx/provisioner_kind.go

```diff
@@ -74,8 +79,22 @@ func Provision(ctx *TestContext) (func(), error) {
 		cluster.ProviderWithLogger(kindLogAdapter{ctx}),
 	)
 	name := fmt.Sprintf("kind-%s", rand.String(16))
+
+	// create cluster with configuration:
+	// 1. use local docker registry
+	// 2. wait for 5 minutes for control plane
+	// 3. use explicit kube config path
+	registry := "kind-registry"
+	// registry := "localhost"
+	registryPort := 5000
 	if err := provider.Create(
 		name,
+		cluster.CreateWithV1Alpha4Config(&v1alpha4.Cluster{
+			ContainerdConfigPatches: []string{
+				fmt.Sprintf(`[plugins."io.containerd.grpc.v1.cri".registry.mirrors."localhost:%[2]d"]
```

Comment:
this is interesting...what does this do exactly? It creates the cluster and mirrors content from a registry outside the cluster to be on all the kind "nodes" within the cluster? Does it continuously sync as new images are pushed to the external registry?

Reply:
This configures the kind cluster configuration to use the local registry.  See these pages:

https://kind.sigs.k8s.io/docs/user/local-registry/
https://github.com/containerd/containerd/blob/master/docs/cri/registry.md#configure-registry-endpoint

Note that the current configuration in the PR uses `http` since that's how the registry is currently setup.